### DISCUSSION TITLE
Where vs Join performance

### DISCUSSION THREAD

**Discussion by pabloBelt** [agershun/alasql#1638] [General]
Hi,

Where conditions implementation are inefficient in AlaSQL compared to Join conditions.

`select * from X, Y where Y.B = X.A and Y.C > 100`
vs
`select * from X join Y on  Y.B = X.A and Y.C > 100`

**Idea**: Should where conditions converted into Join conditions in some way to reuse the exisintg Join implementation. Is that feasible in AlaSQL?

Regards,
Pablo

**Comment by pabloBelt** [comment]
It seems that there is already something in regard to this. I'll have a look at

![image](https://user-images.githubusercontent.com/125101677/218231528-d827e733-c500-4f83-a641-bcf2fbeb3949.png)

**Reply by mathiasrw** [reply]
Go for it!
